Venmo/src/main/java/com/braintreepayments/api/VenmoClient.java

```diff
@@ -428,6 +445,196 @@ private Intent getLaunchIntent(Configuration configuration, String profileId, St
+                break;
+            case BrowserSwitchStatus.SUCCESS:
+                Uri deepLinkUri = browserSwitchResult.getDeepLinkUrl();
+                if (deepLinkUri != null) {
+                    if (deepLinkUri.getPath().contains("success")) {
+                        braintreeClient.sendAnalyticsEvent("pay-with-venmo.browser-switch.success");
+                        String resourceIdFromBrowserSwitch = Uri.parse(String.valueOf(deepLinkUri)).getQueryParameter("resource_id");
+                        String paymentMethodNonceFromBrowserSwitch = Uri.parse(String.valueOf(deepLinkUri)).getQueryParameter("payment_method_nonce");
+                        String usernameFromBrowserSwitch = Uri.parse(String.valueOf(deepLinkUri)).getQueryParameter("username");
+
+                        String resourceIdFromAppSwitch = "";
+                        String paymentMethodNonceFromAppSwitch = "";
+                        String usernameFromAppSwitch = "";
+                        if (resourceIdFromBrowserSwitch == null && paymentMethodNonceFromBrowserSwitch == null && usernameFromBrowserSwitch == null) {
+                            String cleanedAppSwitchUri = String.valueOf(deepLinkUri).replaceFirst("&","?");
+                            resourceIdFromAppSwitch = Uri.parse(String.valueOf(cleanedAppSwitchUri)).getQueryParameter("resource_id");
+                            paymentMethodNonceFromAppSwitch = Uri.parse(String.valueOf(cleanedAppSwitchUri)).getQueryParameter("payment_method_nonce");
+                            usernameFromAppSwitch = Uri.parse(String.valueOf(cleanedAppSwitchUri)).getQueryParameter("username");
+                        }
+
+                        if (resourceIdFromBrowserSwitch != null || resourceIdFromAppSwitch != null) {
+                            String resourceId = "";
+                            if (resourceIdFromBrowserSwitch != null) {
+                                resourceId = resourceIdFromBrowserSwitch;
+                            } else {
+                                resourceId = resourceIdFromAppSwitch;
+                            }
+                            venmoApi.createNonceFromPaymentContext(resourceId, new VenmoOnActivityResultCallback() {
+
+                                @Override
+                                public void onResult(@Nullable VenmoAccountNonce venmoAccountNonce, @Nullable Exception error) {
+                                    if (venmoAccountNonce != null) {
+                                        callback.onResult(venmoAccountNonce, null);
+                                    } else {
+                                        callback.onResult(null, error);
+                                    }
+                                }
+                            });
+                        } else if ((paymentMethodNonceFromBrowserSwitch != null && usernameFromBrowserSwitch != null) || (paymentMethodNonceFromAppSwitch != null && usernameFromAppSwitch != null)) {
```

Comment:
Not sure if this is overkill but it might be helpful for future understanding to pull these if statements into helper methods too that say what these conditions mean - I'm reading this as if `returnToAppSuccessWithPaymentContext()` else if `returnToAppSuccessWithoutPaymentContext()` is that right?

Reply:
Yeah, we could do that, I think this logic is mostly similar to the logic in `onVenmoResult` ([here](https://github.com/braintree/braintree_android/blob/fdd6bdc23e5a9c076ff800939d79096c9f64a0dc/Venmo/src/main/java/com/braintreepayments/api/VenmoClient.java#L247-L296)). Agree it's a bit hard to reason about so if we think it'd be more clear to extract into a helper method I am down.